piper.in

```diff
@@ -7,6 +7,10 @@ import sys
 gi.require_version('Gio', '2.0')
 gi.require_version('Gtk', '3.0')
 from gi.repository import Gio, Gtk
+gtk_version = Gtk.get_major_version(), Gtk.get_minor_version()
+if gtk_version < (3, 22):
```

Comment:
oh, right. yeah, that's doable. I though you meant moving the detection into meson altogether. Want to take this? I'm currently drowning in an email backlog :)